packages/react-native/ReactCommon/react/nativemodule/core/platform/ios/ReactCommon/RCTTurboModule.mm

```diff
@@ -340,12 +387,51 @@ id convertJSIValueToObjCObject(jsi::Runtime &runtime, const jsi::Value &value, s
       TurboModulePerfLogger::asyncMethodCallExecutionStart(moduleName, methodNameStr.c_str(), asyncCallCounter);
     }
 
-    @try {
-      [inv invokeWithTarget:strongModule];
-    } @catch (NSException *exception) {
-      throw convertNSExceptionToJSError(runtime, exception);
-    } @finally {
-      [retainedObjectsForInvocation removeAllObjects];
+    NSError *caughtException = nil;
+    BOOL shouldCatchException = isSync || reject;
+    try {
+      @try {
+        [inv invokeWithTarget:strongModule];
+       } @catch (NSException *exception) {
+         caughtException = maybeCatchException(
+                     shouldCatchException, @{
+                       @"name": exception.name,
+                       NSLocalizedDescriptionKey: exception.reason,
+                       @"stackSymbols": exception.callStackSymbols,
+                       @"stackReturnAddresses": exception.callStackReturnAddresses,
+                     });
```

Comment:
Maybe you can handle `isKindOfClass: [NSException class]` inside `maybeCatchException`?
```suggestion
         caughtException = maybeCatchException(shouldCatchException, exception);
```

Reply:
Thanks, yes, that makes the catch much cleaner.